src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8622,7 +8647,28 @@ internal static List<AlternateStreamData> GetStreams(string path)
             SafeFindHandle handle = NativeMethods.FindFirstStreamW(
                 path, NativeMethods.StreamInfoLevels.FindStreamInfoStandard,
                 findStreamData, 0);
-            if (handle.IsInvalid) throw new Win32Exception();
+            if (handle.IsInvalid)
+            {
+                int error = Marshal.GetLastWin32Error();
+
+                // Directories don't normally have alternate streams, so this is not an exceptional state.
+                // If a directory has no alternate data streams, FindFirstStreamW returns ERROR_HANDLE_EOF.
+                if (error == NativeMethods.ERROR_HANDLE_EOF)
+                {
+                    return alternateStreams;
+                }
+
+                // Filesystems other than NTFS generally don't support alternate streams, and the system
+                // sets ERROR_INVALID_PARAMETER in that case.
+                if (error == NativeMethods.ERROR_INVALID_PARAMETER)
+                {
+                    throw new NotSupportedException("This filesystem does not support alternate data streams.");
```

Comment:
The message string should be in Resx file for localization.

Reply:
This should be in a push I made yesterday. If it's not, it'll be in the next push.